### PR TITLE
Refactor | Test refactoring

### PR DESCRIPTION
**Pull request by MatiasOlivera**
Make the code clearer and readable

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -14,23 +14,18 @@ const rimraf = promisify(_rimraf);
...
-				node =>
-					`${indent}${node.name}\n${
-						node.type === 'directory'
-							? printTree(node.children, indentLevel + 1)
-							: ''
-					}`,
-			),
+			.map(node => {
+				const isDir = node.type === 'directory';
+				const dirTree = isDir ? printTree(node.children, indentLevel + 1) : '';
```
Comment (Andarist) [inline test/index.test.js]: imho this would be better inlined, after all its not a `dirTree` in `!isDir` case

File: test/index.test.js [hunk 2]
```diff
@@ -60,13 +55,24 @@ describe('fixtures', () => {
...
-				));
-			} catch (err) {}
-			script = script || DEFAULT_SCRIPT;
+			async function getBuildScript() {
+				try {
+					const pkgJSON = await readFile(
+						resolve(fixturePath, 'package.json'),
+						'utf8',
+					);
+					const pgk = JSON.parse(pkgJSON);
```
Comment (Andarist) [inline test/index.test.js]: typo, should be `pkg`

File: test/index.test.js [hunk 3]
```diff
@@ -60,13 +55,24 @@ describe('fixtures', () => {
...
-			let script;
-			try {
-				({ scripts: { build: script } = {} } = JSON.parse(
-					await readFile(resolve(fixturePath, 'package.json'), 'utf8'),
-				));
-			} catch (err) {}
-			script = script || DEFAULT_SCRIPT;
+			async function getBuildScript() {
+				try {
+					const pkgJSON = await readFile(
```
Comment (Andarist) [inline test/index.test.js]: actually would be better to check if this file exists before reading it unconditionally - that way lack of package.json could be handled more explicitly here

File: test/index.test.js [hunk 4]
```diff
@@ -60,13 +55,24 @@ describe('fixtures', () => {
...
 			await rimraf(resolve(`${fixturePath}/.rts2_cache_umd`));
 
-			let script;
-			try {
-				({ scripts: { build: script } = {} } = JSON.parse(
-					await readFile(resolve(fixturePath, 'package.json'), 'utf8'),
-				));
-			} catch (err) {}
-			script = script || DEFAULT_SCRIPT;
+			async function getBuildScript() {
```
Comment (Andarist) [inline test/index.test.js]: personally i would prefer adding a parameter to this and hoisting it to the top level scope

File: test/index.test.js [hunk 5]
```diff
@@ -60,13 +55,24 @@ describe('fixtures', () => {
...
+					return build;
+				} catch (err) {
+					// Pass the no-pkg test
+					// because it does not have the package.json file
+					return DEFAULT_SCRIPT;
+				}
+			}
+
+			const build = await getBuildScript();
+			const script = build || DEFAULT_SCRIPT;
```
Comment (Andarist) [inline test/index.test.js]: defaulting should be handled by `getBuildScript`

File: test/index.test.js (hunk 1 above)
Comment (MatiasOlivera) [reply test/index.test.js]: You are right, I will change it.

File: test/index.test.js (hunk 3 above)
Comment (MatiasOlivera) [reply test/index.test.js]: Yes, that's better

File: test/index.test.js (hunk 5 above)
Comment (MatiasOlivera) [reply test/index.test.js]: That makes sense to me

File: test/index.test.js [hunk 6]
```diff
@@ -14,26 +14,46 @@ const rimraf = promisify(_rimraf);
...
+			return defaultScript;
+		}
+	}
+
+	try {
+		const pkgJSON = await readFile(pkgJsonPath, 'utf8');
+		const pkg = JSON.parse(pkgJSON);
+		const { scripts: { build } = {} } = pkg;
+
+		// Return default script if no have
```
Comment (Andarist) [inline test/index.test.js]: imho this comment is excessive, what `return build || defaultScript` does is pretty obvious without it

File: test/index.test.js [hunk 7]
```diff
@@ -14,26 +14,46 @@ const rimraf = promisify(_rimraf);
...
+
+	try {
+		await stat(pkgJsonPath);
+	} catch (err) {
+		if (err.code === 'ENOENT') {
+			return defaultScript;
+		}
+	}
+
+	try {
```
Comment (Andarist) [inline test/index.test.js]: is this try/catch needed now?

File: test/index.test.js [hunk 8]
```diff
@@ -14,26 +14,46 @@ const rimraf = promisify(_rimraf);
...
+				}`;
+			}),
 	);
 };
 
+const getBuildScript = async (fixturePath, defaultScript) => {
+	const pkgJsonPath = resolve(fixturePath, 'package.json');
+
+	try {
+		await stat(pkgJsonPath);
```
Comment (Andarist) [inline test/index.test.js]: according to - https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback 
> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

Comment (MatiasOlivera) [reply test/index.test.js]: Yes, I had also read it. What do we do?

File: test/index.test.js (hunk 6 above)
Comment (MatiasOlivera) [reply test/index.test.js]: Yes, I don't know why I added it

File: test/index.test.js (hunk 7 above)
Comment (MatiasOlivera) [reply test/index.test.js]: IMHO I think it makes the intention clear, anyway I am open to suggestions to improve it

File: test/index.test.js (hunk 8 above)
Comment (developit) [reply test/index.test.js]:
```js
let pkg = {};
try {
  pkg = JSON.parse(await readFile(pkgJsonPath, 'utf8'));
} catch (e) {}
return pkg && pkg.scripts && pkg.scripts.build;

File: test/index.test.js [hunk 9]
```diff
@@ -14,26 +14,32 @@ const rimraf = promisify(_rimraf);
...
 	);
 };
 
+const getBuildScript = async (fixturePath, defaultScript) => {
+	let pkg = {};
+	try {
+		pkg = JSON.parse(
+			await readFile(resolve(fixturePath, 'package.json'), 'utf8'),
+		);
+	} catch (e) {}
```
Comment (Andarist) [inline test/index.test.js]: shouldnt this check `e.code` and rethrow if its not the error we are looking for?

Comment (ForsakenHarmony) [reply test/index.test.js]: this wasn't done before either

Comment (Andarist) [reply test/index.test.js]: yeah, sure - but this PR is an opportunity to improve the code here, so IMHO it could be improved as part of it

Comment (MatiasOlivera) [reply test/index.test.js]: I think it could save debugging time in the future if there are problems with some package.json

**Pull request merged**